### ISSUE TITLE
Output of solvers

### ISSUE THREAD

**Issue by flgoyens** [JuliaNLSolvers/Optim.jl#1109]
Hello, 

I would like to run simple unconstrained algorithms (like gradient descent, trust-regions etc.) and have the solver return the following quantities: 
- the function value at each iterate visited by the algorithm
- the gradient norm at each iterate
- the time spent in each iteration
- a list of all the iterates x_k
Is this currently supported ? I wasn't able to find this in the documentation. 

Thanks a lot
Florentin

**Comment by pkofod** [comment]
This sounds like the "trace"?  If you write `Optim.trace(your_results)` does it have the information you are looking for?
